test/src/org/apache/jmeter/gui/action/HtmlReportGeneratorSpec.groovy

```diff
@@ -0,0 +1,128 @@
+                JMeterUtils.getResString("generate_report_ui.output_directory") + MessageFormat.format(JMeterUtils.getResString(HtmlReportGenerator.CANNOT_CREATE_DIRECTORY), outputDirectoryPath)
+            ]
+    }
+    
+    def "check that report generation succeeds and statistic are generated"(){
+        setup:
+            File testDirectory = new File(JMeterUtils.getJMeterBinDir(), "/testfiles/testReport")
+            if(testDirectory.exists()) {
+                if (testDirectory.list().length>0) {
+                    FileUtils.cleanDirectory(testDirectory)
+                }
+            } else {
+                testDirectory.mkdir()
+            }
+            ObjectMapper mapper = new ObjectMapper()
+            File expected = new File(JMeterUtils.getJMeterBinDir(),"/testfiles/HTMLReportExpect.json")
+            JsonNode expectedRoot = null;
+            expected.withReader { jsonFileReader ->
+                expectedRoot = mapper.readTree(jsonFileReader)
+            }
+        when:
+            HtmlReportGenerator htmlReportGenerator = new HtmlReportGenerator(
+                    JMeterUtils.getJMeterBinDir() + "/testfiles/HTMLReportTestFile.csv",
+                    JMeterUtils.getJMeterBinDir() + "/user.properties", 
+                    JMeterUtils.getJMeterBinDir() + "/testfiles/testReport")
+            List<String> resultList = htmlReportGenerator.run()
+            File statistics = new File(JMeterUtils.getJMeterBinDir(), "/testfiles/testReport/statistics.json")
+            JsonNode actualRoot = null;
+            if (statistics.exists()) {
+                statistics.withReader { jsonFileReader -> 
+                    actualRoot = mapper.readTree(jsonFileReader)
+                }
+            }
+        then:
+            resultList.isEmpty()
+            statistics.exists()
+            expectedRoot == actualRoot
+        cleanup:
+            if(testDirectory.exists()) {
+                if (testDirectory.list().length>0) {
+                    FileUtils.cleanDirectory(testDirectory)
+                }
+            }
+    }
+    
+    def "check that report generation fails when format does not match and error is reported"(){
+        setup:
+            File testDirectory = new File(JMeterUtils.getJMeterBinDir(),"/testfiles/testReportThatShouldBeEmpty")
+            if(testDirectory.exists()) {
+                if (testDirectory.list().length>0) {
+                    FileUtils.cleanDirectory(testDirectory)
+                }
+            } else {
+                testDirectory.mkdir()
+            }
+        when:
+            HtmlReportGenerator htmlReportGenerator = new HtmlReportGenerator(
+                JMeterUtils.getJMeterBinDir() + "/testfiles/HTMLReportFalseTestFile.csv",
+                JMeterUtils.getJMeterBinDir() + "/user.properties", 
+                JMeterUtils.getJMeterBinDir() + "/testfiles/testReport")
```

Comment:
Apparently `testDirectory.toString()` should be used here rather than `JMeterUtils.getJMeterBinDir() + "/testfiles/testReport"`

Reply:
Indeed